zerver/lib/email_mirror_server.py

```diff
@@ -100,15 +102,40 @@ async def handle_RCPT(
             if not is_missed_message_address(address):
                 # Missed message addresses are one-time use, so we don't need
                 # to worry about emails to them resulting in message spam.
-                recipient_realm = await sync_to_async(
-                    lambda a: decode_stream_email_address(a)[0].realm
-                )(address)
-                rate_limit_mirror_by_realm(recipient_realm)
+                channel_email = await sync_to_async(lambda a: decode_stream_email_address(a)[0])(
+                    address
+                )
+                rate_limit_mirror_by_realm(channel_email.realm)
+
+                # This repeats the access check from
+                # process_stream_message, so we can reject it here for
+                # a better user experience.
```

Comment:
Would it make sense to put that access check into a function that accepts the `channel_email` object, and calling that function from both places? It feels finicky enough that we may regret having two copies.

Reply:
Sure, reasonable to factor it out.